src/diff/index.test.js

```diff
@@ -173,5 +174,15 @@ describe('.diff', () => {
         expect(diff(lhs, rhs)).toEqual({ b: 2 });
       });
     });
+
+    describe('nested NaN', () => {
+      test('returns empty object when there is nested NaN value', () => {
```

Comment:
This test description confused me.

More concrete wording:

```suggestion
      test('treats NaN -> NaN as unchanged', () => {
```